src/payloads/AppUser.ts

```diff
@@ -1,6 +1,43 @@
-import { IsInt } from 'class-validator';
+import { IsInt, IsString, IsEmail, IsEnum } from 'class-validator';
+import { AppUserRole } from '@Entities';
 
 export class AppUserPKPayload {
   @IsInt()
   readonly id: number;
 }
+
+abstract class BaseAppUserPayload {
+  @IsString()
+  readonly firstName: string;
+
+  @IsString()
+  readonly lastName: string;
+
+  @IsEmail()
+  readonly email: string;
+
+  @IsString()
+  readonly major: string;
+}
+
+export class EventSignInRequest extends BaseAppUserPayload {}
```

Comment:
hm thoughts on moving this to event payloads? feels a little weird to have eventreq/res in the appuser file. we can import the appuserpayload in the event file :)

Reply:
Yeah we can def do that